metadata-ingestion/src/datahub/ingestion/source/aws/aws_common.py

```diff
@@ -93,10 +97,20 @@ def get_session(self) -> Session:
             return Session(region_name=self.aws_region, profile_name=self.aws_profile)
 
     def get_s3_client(self) -> "S3Client":
-        return self.get_session().client("s3")
+        return self.get_session().client(
+            "s3",
+            endpoint_url=self.aws_endpoint_url,
+            config=Config(proxies=self.aws_proxy),
+        )
 
     def get_s3_resource(self) -> "S3ServiceResource":
-        return self.get_session().resource("s3")
+        resource = self.get_session().resource(
+            "s3",
+            endpoint_url=self.aws_endpoint_url,
+            config=Config(proxies=self.aws_proxy),
+        )
+        resource.meta.client.meta.events.unregister("before-sign.s3", fix_s3_host)
```

Comment:
Please, can you add some comments on why this needs here?

Reply:
when specifying custom endpoint url boto3 tries to connect to {bucket}.s3.amazonaws.com, which is wrong as we had specified our own endpoint_url to which it should connect. Without this "fix_s3 host" we can only list the buckets but we cannot iterate over them or perform any actions. 

Here is some more in depth view of the problem:
https://stackoverflow.com/questions/32618216/override-s3-endpoint-using-boto3-configuration-file

If you are hesitating about this change I can add additional if statement to perform unregister only when endpoint_url is specified:
if endpoint_url is not None:
    resource.meta.client.meta.events.unregister("before-sign.s3", fix_s3_host)

Reply:
If you could add that extra check and also comment what the StackOverflow link in the code that would be awesome, thanks!

Reply:
you mean whole link or just the :
version 1
```
# boto3 only reads the signature version for s3 from that config file. boto3 automatically changes the endpoint to 
# your_bucket_name.s3.amazonaws.com when it sees fit. If you'll be working with both your own host and s3, you may wish 
# to override the functionality rather than removing it altogether.
```

or perhaps:
version 2
```
# according to: https://stackoverflow.com/questions/32618216/override-s3-endpoint-using-boto3-configuration-file
# boto3 only reads the signature version for s3 from that config file. boto3 automatically changes the endpoint to # your_bucket_name.s3.amazonaws.com when it sees fit. If you'll be working with both your own host and s3, you may wish 
# to override the functionality rather than removing it altogether."
```

which one would you prefer?

Also the if statement will be checking for `endpoint_url is not None` only as the proxies are not connected to this unregister function, unless you would prefer it to do both checks for url and proxy together - let me know! : )

Reply:
@treff7es  I added the newest commit with requested changes, could You take a look?